frontend/pages/hosts/details/cards/Software/SelfService/SelfServiceItem/SelfServiceItem.tsx

```diff
@@ -53,18 +54,28 @@ interface IInstallerInfoProps {
 }
 
 const InstallerInfo = ({ software }: IInstallerInfoProps) => {
-  const { name, source, software_package: installerPackage } = software;
+  const {
+    name,
+    source,
+    software_package: installerPackage,
+    app_store_app: vppApp,
+  } = software;
   return (
     <div className={`${baseClass}__item-topline`}>
       <div className={`${baseClass}__item-icon`}>
-        <SoftwareIcon name={name} source={source} size="large" />
+        <SoftwareIcon
+          url={vppApp?.icon_url}
+          name={name}
+          source={source}
+          size="large"
+        />
       </div>
       <div className={`${baseClass}__item-name-version`}>
         <div className={`${baseClass}__item-name`}>
           {name || installerPackage?.name}
         </div>
         <div className={`${baseClass}__item-version`}>
-          {installerPackage?.version || ""}
+          {installerPackage?.version ?? vppApp?.version ?? ""}
```

Comment:
Are you sure you want a nullish check (`??`) rather than a falsey check (`||`) here? The nullish check will only proceed to the second value of the expression if the first value is `null` or `undefined` (e.g., if the first value for `version` is an empty string but the second value is "1.2.3", it the nullish check will yield the empty string).

At first glance, it is probably ok as you have it in this context (assuming we can trust the API typing), but I think it is always good to explicitly consider the possibility since potential bugs would be very subtle. 

There's certainly drawbacks and traps for the unwary with falsey checks too in javascript given all its quirks with type coercion. And I don't know that there's an easy rule of thumb either way. Personally, I tend treat falsey checks as the more idiomatic javascript unless I have a specific reason to prefer the other (e.g., I might use a nullish check for something like a count where we want to explicitly display a zero value from the response object rather than some fallback string).